code/modules/reagents/chemistry/holder.dm

```diff
@@ -89,6 +142,348 @@
 		//Using IDs because SOME chemicals (I'm looking at you, chlorhydrate-beer) have the same names as other chemicals.
 	return english_list(data)
 
+/////////////////////////////////////////////////////////////////////////////////
+///////////////////////////UI / REAGENTS LOOKUP CODE/////////////////////////////
+/////////////////////////////////////////////////////////////////////////////////
+
+
+/datum/reagents/ui_interact(mob/user, datum/tgui/ui)
+	ui = SStgui.try_update_ui(user, src, ui)
+	if(!ui)
+		ui = new(user, src, "Reagents", "Reaction search")
+		ui.status = UI_INTERACTIVE //How do I prevent a UI from autoclosing if not in LoS
+		ui_tags_selected = NONE //Resync with gui on open (gui expects no flags)
+		ui_reagent_id = null
+		ui_reaction_id = null
```

Comment:
This will handle multiple users poorly

Reply:
How so? I haven't seen any issues with it on other codebases

Reply:
Because it will reset the UI for everyone if someone else who doesn't have the UI open opens it

Reply:
Resolved